### PR TITLE
Try to be more consistent about deleting test apps in devicelab logic

### LINKED ISSUE #137555: Android/iOS tests not cleaning up applications after test
Some tests are not cleaning up applications after running tests causing the devices to eventually run out of space. The packages not being deleted are:

package:com.example.image_list
package:com.yourcompany.integration_ui
package:com.example.app
package:com.example.platform_channels_benchmarks

### PR DESCRIPTION
**Pull request by Hixie**
Fixes #137555.

This is an updated version of https://github.com/flutter/flutter/pull/146856, which was reverted in https://github.com/flutter/flutter/pull/146927. The first commit is identical to the original PR, and subsequent commits are the fixes to address failures detected in devicelab post-commit.

### CODE REVIEW

File: dev/devicelab/lib/tasks/platform_channels_benchmarks.dart [hunk 1]
```diff
@@ -39,15 +39,13 @@ TaskFunction runTask(adb.DeviceOperatingSystem operatingSystem) {
         '-d',
         device.deviceId,
       ];
-      return utils.startFlutter(
+      final Process flutterProcess = await utils.startFlutter(
         'run',
         options: options,
       );
+      final Map<String, double> results = await microbenchmarks.readJsonResults(flutterProcess);
```
Comment (reidbaker) [inline dev/devicelab/lib/tasks/platform_channels_benchmarks.dart:46]: I assume microbenchmarks.readJsonResults can work from a different directory but I dont actually know.

**Review by reidbaker — APPROVED** [review]

File: dev/devicelab/lib/tasks/platform_channels_benchmarks.dart (hunk 1 above)
Comment (Hixie) [reply dev/devicelab/lib/tasks/platform_channels_benchmarks.dart:46]: as far as i can tell it works entirely from the provided process' stdout and has no dependency on the current working directory.

Comment (reidbaker) [reply dev/devicelab/lib/tasks/platform_channels_benchmarks.dart:46]: That works for me. Thanks for taking this bug :)

**Comment by Hixie** [comment]
Reason for revert: more failures

**Pull request merged**